packages/next/README.md

```diff
@@ -815,6 +815,9 @@ Router.events.on('routeChangeError', (err, url) => {
 })
 ```
 
+> **Note**: Using router events in `getInitialProps` is discouraged as it may result in unexpected behavior.<br/>
```

Comment:
I think this needs example code using `useEffect`

Reply:
https://github.com/zeit/next.js/pull/7745